tests/management/test_resource_manager.py

```diff
@@ -1260,3 +1262,253 @@ def test_threaded_initialize(self):
+        self.assertTrue(resource.is_available(), "Expected available "
+                        "complex resource with name %r in DB, found %d"
+                        % (self.COMPLEX_NAME, resources_num))
+
+        request = ResourceRequest('res1', AlterDemoComplexResource,
+                                  name=self.COMPLEX_NAME)
+
+        resources = self.client.request_resources(requests=[request]).values()
+
+        resources_num = len(resources)
+        self.assertEquals(resources_num, 1, "Expected list with 1 "
+                          "resource in it but found %d" % resources_num)
+
+        resource, = resources
+        self.assertEquals(resource.name, self.COMPLEX_NAME,
+                          "Expected resource with name %r but got %r"
+                          % (self.COMPLEX_NAME, resource.name))
+
+        self.assertIsInstance(resource, request.type,
+                              "Expected resource of type %r, but got %r"
+                              % (request.type.__name__,
+                                 resource.__class__.__name__))
+
+        self.assertEquals(len(list(resource.get_sub_resources())), 2,
+                          "Expected to have 2 sub-resources, found %r"
+                          % resource.get_sub_resources())
+
+        self.assertTrue(resource.data.initialization_flag,
+                        "Resource %r should have been initialized" %
+                        resource.name)
+
+        for sub_resource in resource.get_sub_resources():
+            self.assertFalse(sub_resource in AlterDemoComplexResource.__dict__,
+                             "Sub-resource %r is still a placeholder" %
+                             sub_resource.name)
+
+            sub_data = DemoResourceData.objects.get(name=sub_resource.name)
+            self.assertFalse(sub_data.is_available(),
+                             "Sub-resource %r should be locked but "
+                             "found available" % sub_resource.name)
+
+            self.assertIsInstance(sub_resource, DemoResource,
+                                  "Expected sub-resource of type %r, got %r"
+                                  % (DemoResource.__name__,
+                                     sub_resource.__class__.__name__))
+
+            self.assertTrue(sub_resource.data.initialization_flag,
+                            "Sub-resource %r should have been initialized" %
+                            sub_resource.name)
+
+        resources_data = request.type.DATA_CLASS.objects.filter(~Q(owner=""),
+                                                   name=self.COMPLEX_NAME)
+
+        resources_num = len(resources_data)
+        self.assertEquals(resources_num, 1, "Expected 1 locked "
+                          "resource with name %r in DB, found %d"
+                          % (self.COMPLEX_NAME, resources_num))
+
+        self.client.release_resources(resources=[resource])
+
+    def test_lock_alter_complex_resource_with_service(self):
```

Comment:
What does the alter means?
Change to something clearer...

Reply:
it means "alternative way of making a resource complex"

Reply:
do you have any recommendation?
if not, the docstring would have to do

Reply:
either ``composed`` as the resource is composed of its fields
or the full word ``alternative``